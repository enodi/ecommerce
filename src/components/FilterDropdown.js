import React from "react";

const FilterDropdown = () => {
  return (
    <select className="filter-by" defaultValue="filter">
      <option disabled value="filter">
        Filter By
      </option>
      <option value="all">All Products</option>
      <option value="new">New Products</option>
      <option value="sets">Sets</option>
      <option value="skincare">Skincare</option>
      <option value="hair">Hair & Body Care</option>
    </select>
  );
};

export default FilterDropdown;
