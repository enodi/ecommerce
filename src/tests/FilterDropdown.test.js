import React from "react";
import { shallow } from "enzyme";

import FilterDropdown from "../components/FilterDropdown";

let wrapper;
beforeEach(() => {
  wrapper = shallow(<FilterDropdown />);
});

test("should render FilterDropdown component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should find first option rendered on dropdown", () => {
  expect(wrapper.find("option").first().text()).toEqual("Filter By");
});
