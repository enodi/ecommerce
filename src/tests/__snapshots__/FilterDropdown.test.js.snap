// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FilterDropdown component correctly 1`] = `
<select
  className="filter-by"
  defaultValue="filter"
>
  <option
    disabled={true}
    value="filter"
  >
    Filter By
  </option>
  <option
    value="all"
  >
    All Products
  </option>
  <option
    value="new"
  >
    New Products
  </option>
  <option
    value="sets"
  >
    Sets
  </option>
  <option
    value="skincare"
  >
    Skincare
  </option>
  <option
    value="hair"
  >
    Hair & Body Care
  </option>
</select>
`;
